lib/control/src/pneumatics.rs

```diff
@@ -0,0 +1,193 @@
+            },
+            orange: match self.state.pink_orange {
+                PinkOrangeState::PinkOnOrangeOff => DigitalSignal::Low,
+                PinkOrangeState::PinkOffOrangeOn => DigitalSignal::High,
+            },
+            green: match self.state.green {
+                SolenoidSignal::Off => DigitalSignal::Low,
+                SolenoidSignal::On => DigitalSignal::High,
+            },
+            blue_1: match self.state.blue {
+                SolenoidSignal::Off => DigitalSignal::Low,
+                SolenoidSignal::On => DigitalSignal::High,
+            },
+            blue_2: match self.state.blue {
+                SolenoidSignal::Off => DigitalSignal::Low,
+                SolenoidSignal::On => DigitalSignal::High,
+            },
+        }
+    }
+
+    /// Set the state of the yellow solenoid
+    pub fn set_yellow(&mut self, signal: SolenoidSignal) {
+        self.state.yellow = signal;
+    }
+
+    /// Set the state of the pink and orange solenoids
+    pub fn set_pink_orange(&mut self, state: PinkOrangeState) {
+        self.state.pink_orange = state;
+    }
+
+    /// Set the state of the green solenoid
+    pub fn set_green(&mut self, signal: SolenoidSignal) {
+        self.state.green = signal;
+    }
+
+    /// Set the state of the blue solenoids
+    pub fn set_blue(&mut self, signal: SolenoidSignal) {
+        self.state.blue = signal;
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_pneumatics_control() {
+        let mut control = PneumaticsControl::new();
+
+        control.set_yellow(SolenoidSignal::On);
+        control.set_pink_orange(PinkOrangeState::PinkOnOrangeOff);
+        control.set_green(SolenoidSignal::On);
+        control.set_blue(SolenoidSignal::On);
+
+        assert_eq!(control.get_state().yellow, SolenoidSignal::On);
+        assert_eq!(
+            control.get_state().pink_orange,
+            PinkOrangeState::PinkOnOrangeOff
+        );
+        assert_eq!(control.get_state().green, SolenoidSignal::On);
+        assert_eq!(control.get_state().blue, SolenoidSignal::On);
+
+        let gpio_out = control.get_gpio_out();
+        assert_eq!(gpio_out.yellow, DigitalSignal::High);
+        assert_eq!(gpio_out.pink, DigitalSignal::High);
+        assert_eq!(gpio_out.orange, DigitalSignal::Low);
+        assert_eq!(gpio_out.green, DigitalSignal::High);
+        assert_eq!(gpio_out.blue_1, DigitalSignal::High);
+        assert_eq!(gpio_out.blue_2, DigitalSignal::High);
+    }
+}
+
+enum BrakeState {
+    Engaged,
+    Disengaged,
+}
+
+enum LateralSuspensionState {
+    Deployed,
+    Retracted,
+}
+
+struct Pneumatics<P: GpioOutputPin> {
+    brakes: BrakeState,
+    lateral_suspension: LateralSuspensionState,
+    brake_pin: P,
+    lateral_suspension_pin: P
+}
+
+impl<P: GpioOutputPin> Pneumatics<P> {
+    fn new(mut brake_pin: P, mut lateral_suspension_pin: P) -> Self {
+        brake_pin.set_high();

```

Comment:
The brakes are actually electrically failsafe, so they are engaged when not powered. (So set the pin to high to disengage and low to engage.) Would be good to document this behaviour in a Rust doc comment for `Pneumatics`